packages/ui-components/src/lib/providers/wallet/WalletProvider.svelte

```diff
@@ -0,0 +1,28 @@
+<script context="module" lang="ts">
+	export const ACCOUNT_KEY = 'account_key';
+	export const USE_ACCOUNT_KEY = 'use_account_key';
+	export type UseAccountResult = {
+		account: Readable<string | null>;
+	};
+	export type UseAccount = () => UseAccountResult;
+</script>
+
+<script lang="ts">
+	import { setContext } from 'svelte';
+	import { readable, type Readable } from 'svelte/store';
+
+	export let account: Readable<string | null> = readable(null);
+
+	// Set the account address store in the context
+	setContext(ACCOUNT_KEY, account);
+
+	const useAccount: UseAccount = () => {
+		return {
+			account: account
+		};
+	};
+	// Set the useAccount function in the context
+	setContext(USE_ACCOUNT_KEY, useAccount);
```

Comment:
why do we need to do this? the `useAccount` function isn't context dependent, what it returns is.